cli/src/pcluster/config/cluster_config.py

```diff
@@ -971,6 +970,27 @@ def _register_validators(self, context: ValidatorContext = None):  # noqa: D102
             self._register_validator(InstanceProfileValidator, instance_profile_arn=self.instance_profile)
 
 
+class HeadNodeAndQueueIam(_BaseIam):
```

Comment:
We can simply call it `Iam` instead of `HeadNodeAndQueueIam` to avoid confusing the name with what it does. Essentially, it's used in either the HeadNode or the Queue but the term `HeadNodeAndQueueIam` might give the impression that the IAM values apply to both the head node and compute nodes at the same time (which is not the case).